apps/server/src/modules/task/service/submission.service.ts

```diff
@@ -20,10 +30,38 @@ export class SubmissionService {
 		return submissions;
 	}
 
+	async findByUserAndTask(userId: EntityId, taskId: EntityId): Promise<Submission[]> {
+		const [submissions] = await this.submissionRepo.findAllByUserId(userId);
+		const filteredSubmissions = submissions.filter((submission) => submission.task.id === taskId);
+
+		return filteredSubmissions;
+	}
+
 	async delete(submission: Submission): Promise<void> {
 		const params = FileParamBuilder.build(submission.school.id, submission);
 		await this.filesStorageClientAdapterService.deleteFilesOfParent(params);
 
 		await this.submissionRepo.delete(submission);
 	}
+
+	async createForTaskCard(userId: EntityId, taskId: EntityId): Promise<Submission> {
+		const user = await this.authorizationService.getUserWithPermissions(userId);
```

Comment:
authorization part should normally be in the uc, even if in this case it's task-card module